internal/store/store.go

```diff
@@ -390,6 +420,17 @@ func (bs *BlockStore) PruneBlocks(height int64, state sm.State) (uint64, int64,
 	if err != nil {
 		return 0, -1, err
 	}
+	bs.blocksDeleted += int64(pruned)
+
+	if bs.compact && bs.blocksDeleted >= bs.compactionInterval {
+		// Error on compaction should not be reason to halt the chain
+		// TODO add logger to keep track of error
+		err = bs.db.Compact(nil, nil)
+		if err != nil && bs.logger != nil {
+			bs.logger.Error("Error compacting block store", "err", err)
```

Comment:
same comment as above, returning an error here I don't think will cause any issues because it's just logged